File: backend/app/schemas/notification.py
"""
RetailFlow AI — Notification Pydantic Schemas
"""

import uuid
from datetime import datetime

from pydantic import BaseModel, Field


class NotificationResponse(BaseModel):
    """Single notification as returned in the inbox."""

    id: uuid.UUID
    alert_config_id: uuid.UUID
    store_id: uuid.UUID
    counter_id: uuid.UUID | None
    user_id: uuid.UUID
    title: str
    message: str
    trigger_value: int
    threshold: int
    alert_type: str
    is_read: bool
    read_at: datetime | None
    email_sent: bool
    created_at: datetime

    model_config = {"from_attributes": True}


class NotificationInboxResponse(BaseModel):
    """Paginated notification inbox for the current user."""

    notifications: list[NotificationResponse]
    total: int
    unread_count: int
    page: int
    limit: int

File: frontend/src/types/notification.ts
/**
 * RetailFlow AI — Notification Types
 * Mirrors backend app/schemas/notification.py
 */

export type NotificationSeverity = "info" | "warning" | "critical";

export interface NotificationItem {
  id: string;
  user_id: string;
  store_id: string | null;
  counter_id: string | null;
  title: string;
  message: string;
  severity: NotificationSeverity;
  is_read: boolean;
  read_at: string | null;
  created_at: string;
}

export interface NotificationInboxResponse {
  notifications: NotificationItem[];
  total: number;
  unread_count: number;
  page: number;
  limit: number;
}

export interface UnreadCountResponse {
  unread_count: number;
  user_id: string;
}
